### PR TITLE
Run production tests on npm package

### LINKED ISSUE #3698: Improve distribution tests
`yarn test:dist` runs the tests against the built JS files, but it didn't catch an issue in 1.10 where a file with path `../../index` couldn't be resolved, because in the `dist` directory, `../../index` resolves to the original source. We could fix this by copying the files to a tempdir before running the tests.

It would be nice to test the actual `.tgz` file, too.

https://github.com/prettier/prettier/commit/cf511ec2e973ed6d794c5b99321f2b3a3458a919

#3697

### PR DESCRIPTION
**Pull request by azz**
Runs the production tests agains the result of actually installing the package, rather than from the `dist` directory. 

Extracted from #3712
Fixes #3698

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,12 +6,10 @@ node_js:
 cache:
   yarn: true
   directories:
-  - node_modules
+    - node_modules
```
Comment (ikatyang) [inline .travis.yml]: 😉

File: .travis.yml [hunk 2]
```diff
@@ -6,12 +6,10 @@ node_js:
...
 env:
   - NODE_ENV=development
   - NODE_ENV=production
-matrix:
-  fast_finish: true
```
Comment (ikatyang) [inline .travis.yml]: ❓

Comment (josephfrazier) [reply .travis.yml]: Yeah, shouldn't we leave this, in order to learn about build failures faster? From https://blog.travis-ci.com/2013-11-27-fast-finishing-builds:

> With fast finishing enabled, Travis CI will mark your build as finished as soon as one of two conditions are met: The only remaining jobs are allowed to fail, or a job has already failed.

We don't have any [jobs allowed to fail](https://docs.travis-ci.com/user/customizing-the-build/#Rows-that-are-Allowed-to-Fail), so it should be safe to leave `fast_finish` enabled.

Comment (azz) [reply .travis.yml]: Ah, I was under the impression it only applied when jobs were allowed to fail. Reverted.

**Comment by duailibe** [comment]
How likely it is that a build will pass in production but fail in development?

**Comment by azz** [comment]
_Shouldn't_ be possible, but I don't know for sure.

**Pull request merged**